packages/zwave-js/src/lib/commandclass/UserCodeCC.ts

```diff
@@ -236,6 +236,48 @@ function validateCode(code: string, supportedChars: string): boolean {
 	return [...code].every((char) => supportedChars.includes(char));
 }
 
+function setUserCodeMetadata(this: UserCodeCC, userId: number) {
+	const valueDB = this.getValueDB();
+	const statusValueId = getUserIdStatusValueID(this.endpointIndex, userId);
+	const codeValueId = getUserCodeValueID(this.endpointIndex, userId);
+	const supportedUserIDStatuses =
+		valueDB.getValue<UserIDStatus[]>(
+			getSupportedUserIDStatusesValueID(this.endpointIndex),
+		) ??
+		(this.version === 1
+			? [
+					UserIDStatus.Available,
+					UserIDStatus.Enabled,
+					UserIDStatus.Disabled,
+			  ]
+			: [
+					UserIDStatus.Available,
+					UserIDStatus.Enabled,
+					UserIDStatus.Disabled,
+					UserIDStatus.Messaging,
+					UserIDStatus.PassageMode,
+			  ]);
+	if (!valueDB.hasMetadata(statusValueId)) {
+		valueDB.setMetadata(statusValueId, {
+			...ValueMetadata.Number,
+			label: `User ID status (${userId})`,
+			states: enumValuesToMetadataStates(
+				UserIDStatus,
+				supportedUserIDStatuses,
+			),
+		});
+	}
+	const codeMetadata: ValueMetadata = {
+		...ValueMetadata.String,
```

Comment:
This needs to distinguish the type based on the actual user code (and default to string if none was passed)